src/data/data.cc

```diff
@@ -146,6 +146,103 @@ void MetaInfo::SetInfo(const char* key, const void* dptr, DataType dtype, size_t
   }
 }
 
+DMatrix::DMatrix(data::SimpleDMatrix* simple_dmat)
+    : simple_dmat_(simple_dmat), in_memory_(true) {}
+
+#if DMLC_ENABLE_STD_THREAD
```

Comment:
This can be removed now since we are requiring c++11.

Reply:
Thanks, removed.

I didn't want to make this PR too big. Will send a separate PR to show the next steps.